### PR TITLE
Fetch documentation from kyma repository on building time website

### LINKED ISSUE #213: Fetch documentation from kyma repository on building time website
**Description**
At the moment we have documentation for all version in `static/documentation` folder. This is unnecessary, and causes `Kyma-bot` top of contributor. 

AC:
- create stage for fetching documentation on time building website on one of Gatsby's lifecycles - the best will be `OnPreBuild` or suggest something better.
- create mocks of documentation for development process.
- remove all generated documentation version from sources of website.

**Reasons**
- Eliminate `Kyma-bot` from contributors 😈

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Fetch documentation from kyma repository on building time website:
  - remove necessity of pushing docs from core repository before publishing new version of documentation
  - eliminate `kyma-bot` from contributors
 
**Related issue(s)**
Resolves #213

### CODE REVIEW

File: tools/content-loader/src/prepare-docs/copy-docs.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    let files;
+    [err, files] = await to(this.getFilesPaths(docsDir));
+    if (err) throw new VError(err, `while getting files paths`);
+
+    const allowedFilesRegex = /docs\/(manifest\.(yaml|yml)|[A-z0-9-_]*\/(docs\.config\.json|docs\/assets\/[A-z0-9-_.]*\.(png|jpg|gif|jpeg|svg|yaml|yml|json)|docs\/[A-z0-9-_.]*\.md))/;
+
+    files = files.filter(file => Boolean(allowedFilesRegex.exec(file)));
+
+    const copyingFiles = files.map(async file => {
```
Comment (aerfio) [inline tools/content-loader/src/prepare-docs/copy-docs.ts]:
```suggestion
    const copiedFiles = files.map(async file => {
```

File: tools/content-loader/src/prepare-docs/copy-docs.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    });
+
+    let result;
+    [err, result] = await to(Promise.all(copyingFiles));
+    if (err) throw new VError(err, `while copying files`);
+
+    return result;
+  };
+
+  private do = async (source: string, output: string) => {
```
Comment (aerfio) [inline tools/content-loader/src/prepare-docs/copy-docs.ts]: _**do**_ find it a better name xd

File: tools/content-loader/.eslinterrc [hunk 3]
```diff
@@ -1,3 +0,0 @@
-{
-  "extends": "react-app"
```
Comment (pkosiec) [inline tools/content-loader/.eslinterrc]: Why did you remove this file?

File: tools/content-loader/src/prepare-docs/copy-docs.ts (hunk 2 above)
Comment (pkosiec) [reply tools/content-loader/src/prepare-docs/copy-docs.ts]: It would be Go (if the type is well named), but in JS I absolutely agree, please be more verbose

File: tools/content-loader/src/prepare-docs/docs-versions.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      writeJSON(outputPath, versions, { encoding: "utf8" }),
+    );
+    if (err)
+      throw new VError(
+        err,
+        `while writing versions ${JSON.stringify(versions)} to ${outputPath}`,
+      );
+  }
+
+  private forReleases(releases?: Map<string, string>) {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/docs-versions.ts]: `generateForReleases`

Rename all methods to be more verbose

File: tools/content-loader/src/prepare-docs/copy-docs.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
+import { join } from "path";
+import { remove, copy, stat, readdir } from "fs-extra";
+import to from "await-to-js";
+import { VError } from "verror";
+
+import GitClient from "../github-client/git-client";
+
+class CopyDocs {
+  releases = async ({ releases, source, output }) => {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/copy-docs.ts]: rename

File: tools/content-loader/src/prepare-docs/copy-docs.ts [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+      [err] = await to(GitClient.checkoutTag(tag));
+      if (err) throw err;
+
+      const out = `${output}/${key}`;
+      [err] = await to(this.do(source, out));
+      if (err) throw new VError(err, `while copying sources from tag: ${tag}`);
+    }
+  };
+
+  branches = async ({ branches, source, output }) => {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/copy-docs.ts]: rename

File: tools/content-loader/src/prepare-docs/docs-versions.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      versions.push({
+        name: name,
+        tag: tag,
+      });
+    });
+
+    return versions;
+  }
+
+  private forBranches(branches?: Map<string, string>) {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/docs-versions.ts]: rename

File: tools/content-loader/src/prepare-docs/tags.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import to from "await-to-js";
+
+import GitHubClient from "../github-client/github-client";
+
+class Tags {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/tags.ts]: do not use plural form, consider for example `TagFetcher`

File: tools/content-loader/src/prepare-docs/releases.ts [hunk 9]
```diff
@@ -1,5 +1,24 @@
...
+import to from "await-to-js";
+import {
+  ReposGetReleaseResponse,
+  ReposListTagsResponseItem,
+} from "@octokit/rest";
+
+import GitHubClient from "../github-client/github-client";
+import { DocsReleasesVersion } from "./docs-versions";
+
+class Releases {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/releases.ts]: do not use plural form, consider for example ReleaseFetcher

File: tools/content-loader/src/prepare-docs/index.ts [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+import { VError } from "verror";
+
+import { CoreConfig } from "../config";
+import docsConfig, { DocsConfig } from "./config";
+import GitClient from "../github-client/git-client";
+import CheckingDocs from "./checking";
+import CopyDocs from "./copy-docs";
+import DocsVersions from "./docs-versions";
+
+export default async (coreConfig: CoreConfig) => {
```
Comment (pkosiec) [inline tools/content-loader/src/prepare-docs/index.ts]: name this function

File: tools/content-loader/src/prepare-docs/branches.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
+import to from "await-to-js";
+import { VError } from "verror";
+
+import GitHubClient from "../github-client/github-client";
+
+class Branches {
```
Comment (akucharska) [inline tools/content-loader/src/prepare-docs/branches.ts]: Rename according to comments from Paweł

File: tools/content-loader/README.md [hunk 12]
```diff
@@ -20,15 +26,15 @@ npm start
...
-| **APP_DOCS_OUTPUT**               | `out`             | The path for storing the documentation results              |
-| **APP_DOCS_VERSIONS_CONFIG_FILE** | `versions.json`   | The path to the website documentation configuration         |
-| **APP_DOCS_TEMP_DIR**             | `tempDir`         | The path for storing temporary data                         |
+| Name                              | Required | Default value     | Description                                                 |
+| ----------------------------------| :------: | :---------------: | ----------------------------------------------------------- |
+| **APP_TOKEN**                     | **YES**  | `null`            | The GitHub API OAuth token                                  |
+| **APP_ORGANIZATION**              | **YES**  | `kyma-project`    | The GitHub organization that owns a given repository        |
+| **APP_REPOSITORY**                | **YES**  | `kyma`            | The repository with documentation                           |
+| **APP_DOCS_BRANCHES**             | **YES**  | `["master"]`      | The branch names with the documentation                     |
```
Comment (mmitoraj) [inline tools/content-loader/README.md]:
```suggestion
| **APP_DOCS_BRANCHES**             | **YES**  | `["master"]`      | The branches with documentation                     |
```

File: tools/content-loader/src/prepare-docs/docs-versions.ts (hunk 4 above)
Comment (magicmatatjahu) [reply tools/content-loader/src/prepare-docs/docs-versions.ts]: I have a rule in tslint to name imported by default things as name of file (in camelCase).

File: tools/content-loader/src/prepare-docs/copy-docs.ts (hunk 5 above)
Comment (magicmatatjahu) [reply tools/content-loader/src/prepare-docs/copy-docs.ts]: I have a rule in tslint to name imported by default things as name of file (in camelCase).

File: tools/content-loader/src/prepare-docs/copy-docs.ts (hunk 6 above)
Comment (magicmatatjahu) [reply tools/content-loader/src/prepare-docs/copy-docs.ts]: I have a rule in tslint to name imported by default things as name of file (in camelCase).

File: tools/content-loader/src/prepare-docs/docs-versions.ts (hunk 7 above)
Comment (magicmatatjahu) [reply tools/content-loader/src/prepare-docs/docs-versions.ts]: I have a rule in tslint to name imported by default things as name of file (in camelCase).
